lib/albacore/task_types/build.rb

```diff
@@ -189,25 +190,10 @@ def heuristic_executable
 
     	  msb = "msbuild_not_found"
     	  maxVersion = -1
-    	  begin
-    		Win32::Registry::HKEY_LOCAL_MACHINE.open('SOFTWARE\Microsoft\MSBuild\ToolsVersions') do |toolsVersion|
-    		  toolsVersion.each_key do |key|
-    			begin
-    			  versionKey = toolsVersion.open(key)
-    			  version = key.to_i
-    			  if maxVersion < version
-    				maxVersion = version
-    				msb = "#{versionKey['MSBuildToolsPath']}\\msbuild.exe"
-    			  end
-    			rescue
-    			  error "failed to open #{key}"
-    			end
-    		  end
-    		end
-    	  rescue
-    		error "failed to open HKLM\\SOFTWARE\\Microsoft\\MSBuild\\ToolsVersions"
-    	  end
-
+        versions = Albacore.find_msbuild_versions
+        if versions.any?
+          msb = versions.max[1]
```

Comment:
Just out of curiosity. Where is this ```max``` method defined. Cannot find it in the ruby docs. But that does not necessarily mean anything. ;)